extensions/amp-ooyala-player/0.1/amp-ooyala-player.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+import {user} from '../../../src/log';
+
+class AmpOoyalaPlayer extends AMP.BaseElement {
```

Comment:
@chibzu for players it's recommended to implement a pausing mechanism so that AMP runtime can pause the video when the user navigates away of the document. For players that don't support a pausing mechanism the only option is to unlayout the player and layout it back again if the user navigates back again - the problem with this is it won't remember where the user left off.

If there's no way to implement pause in the player you need to override `unlayoutOnPause` to return `true`.